File: spring-di-only/commons/src/main/java/com/deploydesexta/springsandbox/services/UserService.java
package com.deploydesexta.springsandbox.services;

import com.deploydesexta.springsandbox.repositories.UserRepository;
import org.springframework.stereotype.Service;

@Service
public class UserService {

    private final UserRepository userRepository;

    public UserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public void greetings() {
        System.out.println("HELLO WORLD");
    }
}

File: bean-qualification-strategies/kt-typealias-injection/src/main/kotlin/services/UserService.kt
package com.deploydesexta.springsandbox.services

import com.deploydesexta.springsandbox.repositories.GetUser
import org.springframework.stereotype.Service

@Service
class UserService(
    private val getUser: GetUser,
) {

    fun greetings() {
        val user = getUser(1L)
        println("HELLO ${user.name}")
    }
}
